python/lsst/pipe/tasks/fit_coadd_multiband.py

```diff
@@ -0,0 +1,321 @@
+# This file is part of pipe_tasks.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = [
+    "CoaddMultibandFitConfig", "CoaddMultibandFitSubConfig", "CoaddMultibandFitSubTask",
+    "CoaddMultibandFitTask",
+]
+
+from .fit_multiband import CatalogExposure
+
+import lsst.afw.table as afwTable
+from lsst.obs.base import ExposureIdInfo
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+import lsst.pipe.base.connectionTypes as cT
+
+import astropy
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Iterable, List, Set
+
+CoaddMultibandFitBaseTemplates = {
+    "name_coadd": "deep",
+    "name_method": "multiprofit",
+}
+
+
+# This class could be merged with CatalogExposureInputs below,
+# but that requires another, uglier workaround; see:
+# https://github.com/python/cpython/issues/83315
+@dataclass(frozen=True, kw_only=True)
+class PsfFitInputs:
+    table_psf_fits: astropy.table.Table
+
+
+@dataclass(frozen=True, kw_only=True)
+class CatalogExposureInputs(CatalogExposure, PsfFitInputs):
+    def get_catalog(self):
+        return self.catalog
+
+
+class CoaddMultibandFitConnections(
+    pipeBase.PipelineTaskConnections,
+    dimensions=("tract", "patch", "skymap"),
+    defaultTemplates=CoaddMultibandFitBaseTemplates,
+):
```

Comment:
Assuming that you are working with deblender models, you'll also need the scarlet models. See
https://github.com/lsst/meas_base/blob/main/python/lsst/meas/base/forcedPhotCoadd.py#L75-L80 and https://github.com/lsst/meas_base/blob/main/python/lsst/meas/base/forcedPhotCoadd.py#L340-L355 for an example of how to do this as of w.2022.26.